system/jlib/jsocket.cpp

```diff
@@ -2691,9 +2704,12 @@ CSocket::CSocket(T_SOCKET new_sock,SOCKETMODE smode,bool _owned)
     accept_cancel_state = accept_not_cancelled;
     set_nagle(false);
     //set_linger(DEFAULT_LINGER_TIME); -- experiment with removing this as closesocket should still endevour to send outstanding data
-#ifdef _TRACE
     char peer[256];
-    peer_name(peer,sizeof(peer));
+    hostport = peer_name(peer,sizeof(peer));
+    hostname = strdup(peer);
+    char lname[256];
+    localPort = name(lname, sizeof(lname), false);
```

Comment:
It may not be an efficient concern, but as per prev. comment (in post_connect) it would look cleaner if method didn't get lname that it didn't need.

Unless it would be useful to store? I notice e.g. logConnectionInfo used to use, but you removed?